src/EntryPoints/MediaWikiHooks.php

```diff
@@ -164,4 +165,13 @@ public static function onArticleRevisionViewCustom(
 		return true;
 	}
 
+	/**
+	 * @param array<string, mixed> $preferences
+	 */
+	public static function onGetPreferences( User $user, array &$preferences ): void {
+		$preferences['wikibase-rdf-acknowledgeedit'] = [
+			'type' => 'api'
+		];
+	}
```

Comment:
Is this dead code? Seems to be about persisting things in user preferences, while storage actually happens in a cookie.

Reply:
No, this is to set a user preference value. Without that the API is not allowed to set the value.